### PR TITLE
docs: Add Color documentation to our foundation section

### PR DESCRIPTION
**Pull request by jhardy**
This PR adds a documentation pages about our color palette and the various groups of color we have.

image

### CODE REVIEW

File: www/src/documentation/foundations/demos/Color.tsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    specializedText: []
+  }
+
+  const colorBands = {
+    core: coreColors.length,
+    intent: coreColors.length + intentColorKeys.length,
+    derivative: coreColors.length + intentColorKeys.length + derivativeColorKeys.length
+  }
+
+  for(const [key, value] of Object.entries(theme.colors)) {
```
Comment (jhardy) [inline www/src/documentation/foundations/demos/Color.tsx:197]: there might be a better way to do this, but the idea here is to break out the theme into their logical grouping, this way we can document each group in a bit more detail

File: www/src/documentation/foundations/demos/Color.tsx [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+const derivativeColorKeys: Array<keyof DerivativeColors> = [
+  'field',
+  'inverse',
+  'inverseOn',
+  'neutral',
+  'linkInteractive',
+  'informAccent',
+  'positiveAccent',
+  'warnAccent',
+]
```
Comment (jhardy) [inline www/src/documentation/foundations/demos/Color.tsx:56]: Declaring color keys here and below feels a bit brittle. I was curious if we should export these from `design-tokens` package. Interested in  thoughts here.

File: www/src/documentation/foundations/demos/Color.tsx [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    }
+    else if(specializedTextColorKeys.includes(key)) {
+      colorList.specializedText.push([key, value])
+    }
+    else {
+      colorList.stateful.push([key, value])
+    }
+  }
+
+  colorList.stateful = chunk(colorList.stateful, 7)
```
Comment (jhardy) [inline www/src/documentation/foundations/demos/Color.tsx:221]: This chunks the stateful colors into logical groupings that made it easer to lay out for the docs

**Review by lukelooker — APPROVED** [review]
@jhardy - I went ahead and hacked on top of your PR to add exports for all of the color groupings this wanted to leverage as well as moving the function to break-up the theme's `color` object into it's requisite parts.

I also moved most of the demo text into `MDX` to help with search discovery and minimize the mixture of code and documentation. 

Assuming tests pass I think this is fine to merge as it doesn't have any "functional" effects on the library outside of improving the documentation around color.

**Review by lukelooker — DISMISSED** [review]
Had to update a single snapshot from theme-editor demo.

**Comment by jhardy** [comment]
@lukelooker thanks for your improvements, much better. I refactored the some of the grids for better responsive handling on the color page. 

I also went ahead and added some docs for the `radii` theme values after I saw you use `theme.radii[3]` and I was confused why it wasn't working, but it needed to be `theme.radii.medium`.

File: www/src/documentation/foundations/demos/ColorList.tsx [hunk 4]
```diff
@@ -41,22 +41,28 @@ const ColorSwatch = styled(({ name, color, ...props }: ColorSwatchProps) => (
   &::before {
     background: ${({ color }) => color};
     border: 1px solid ${({ theme }) => theme.colors.ui3};
-    border-radius: ${({ theme }) => theme.radii[3]};
-    content: ' ';
+    border-radius: ${({ theme }) => theme.radii.medium};
```
Comment (lukelooker) [inline www/src/documentation/foundations/demos/ColorList.tsx:44]: 🤦

File: www/src/documentation/foundations/demos/ColorList.tsx [hunk 5]
```diff
@@ -41,22 +41,28 @@ const ColorSwatch = styled(({ name, color, ...props }: ColorSwatchProps) => (
...
     display: block;
     height: 4rem;
-    width: 7.5rem;
+    margin-bottom: ${({ theme }) => theme.space.xsmall};
+    width: 100%;
   }
 `
 
+const ColorListGrid = styled(Grid)`
+ grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
```
Comment (lukelooker) [inline www/src/documentation/foundations/demos/ColorList.tsx]: I wonder if we should add `autofit` to `Grid` as a prop that takes the minimum value as a string and assumes the max is always `1fr` - seems like a pattern that could/would be useful elsewhere.

**Review by lukelooker — APPROVED** [review]
Pushed a commit with a couple of prettier clean-ups (lint doesn't require them in MDX but it was distracting me).

The custom grids you created expose a nice enhancement we could probably do to the grid component at some point (see inline comment).

**Pull request merged**